jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -101,17 +102,7 @@ private void prepareDirectory() throws IOException {
     }
 
     private File getDestinationFile(Type screenShotType) {
-        if (testContext.contains(AbstractSpringStories.JBEHAVE_SCENARIO)) {
-            String storyName = testContext.get(AbstractSpringStories.JBEHAVE_SCENARIO, String.class).split("#")[0];
-            File destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, screenShotType, storyName));
-            int i = 1;
-            while (destinationFile.exists()) {
-                destinationFile = new File(screenshotDirectory, String.format(FILE_NAME_PATTERN, screenShotType, storyName + "-" + i++));
-            }
-            return destinationFile;
-        } else {
-            return new File("screenshot-" + System.currentTimeMillis() + ".png");
-        }
+        return fileNameResolver.resolveFilePath(FILE_NAME_PATTERN, screenshotDirectory, screenShotType.toString()).toFile();
```

Comment:
We can probably get rid of this method and call fileNameResolver directly from `takesScreenshot`